core/partitioning/shape_analysis.cpp

```diff
@@ -340,6 +340,19 @@ void getSegmentsOutputByRunning(
 
   seg_block.register_inshapes(input_shapes, shape_mode);
   seg_block.register_intypes(input_types);
+
+  // get output type for each segmented block so this can be used in conversion process
+  std::vector<at::ScalarType> output_types;
```

Comment:
One thing to check: Because we're running this at the end of shape propagation does this work for fully convertible modules?

From memory I think that shape prop is only called form BuildHybridGraph->partition() which I don't think is called if all ops are convertible.